### PR TITLE
Make getCustomerCards() return the same type of object in all places

### PR DESCRIPTION
**Pull request by scottcwilson**

### CODE REVIEW

**Comment by scottcwilson** [comment]
Sample log from version 2.2.3

[demo.log](https://github.com/proseLA/authorizenet_cim/files/6325000/demo.log)

File: includes/modules/payment/authorizenet_cim.php [hunk 1]
```diff
@@ -1635,6 +1635,26 @@ function getCustomerCards($customerID, $all = false)
 			return $customer_cards;
 		}
 
+      function getCustomerCards($customerID, $all = false)
+      {
+   	    $cards_on_file =  parent::getCustomerCardList($customerID, $all);
```
Comment (proseLA) [inline includes/modules/payment/authorizenet_cim.php:1640]: we are in _cim here.  parent would be base.  i think this line is incorrect.

**Review by proseLA — CHANGES_REQUESTED** [review]
scott,
i do not like the naming.  i would prefer to leave method getCustomerCards in the _cim class unchanged;  i think the new method getCustomerCardList should be the one that returns the array, and we can add that also into the _cim class.

then we can remove all of it from the _cof class.

we would then need to change the calls on the admin observer and in the modules card update.

does that make sense?

**Comment by proseLA** [comment]
the demo log does not make sense to me.  let me look at the history a bit.

**Comment by scottcwilson** [comment]
Let's close this for now.  You had fixed the issue I was seeing in a more recent version of the software, so these changes don't really make sense - but you may want to consider the change in `admin/includes/classes/observers/class.cim_admin_observer.php`.

**Comment by proseLA** [comment]
yeah.  in the future please open an issue as opposed to doing the work on a PR.  the more i am now looking at this PR, the more problems i am seeing.  there were different elements to the array in _cim `getCreditCards` than in the fields of the object of resultQuery...  and your `!empty` in the template file was set the wrong way.

there is a lot of lines of code here, and at some point i hope to be able to refactor it and make it cleaner, but there are very few people using it; and my affiliate link for auth.net nets me $7.50/month.  barely enough to get extra cheese on my whopper!  

i'm going to do a small refactor as i agree with you on the original point; i have the same method name in different classes which returned different object types.  and i do not like that at all.  despite it working fine.

best.

best.

**Comment by scottcwilson** [comment]
Very good.  Thanks for the review!

**Pull request closed without merging**